avaframe/com1DFAPy/com1DFA.py

```diff
@@ -943,9 +950,16 @@ def DFAIterate(cfg, particles, fields, dem):
             log.info(('cpu time Position = %s s' % (Tcpu['Pos'] / nIter)))
             log.info(('cpu time Neighbour = %s s' % (Tcpu['Neigh'] / nIter)))
             log.info(('cpu time Fields = %s s' % (Tcpu['Field'] / nIter)))
-            Particles.append(copy.deepcopy(particles))
-            Fields.append(copy.deepcopy(fields))
-            nSave = nSave + 1
+            if 'particles' in resTypes:
+                Particles.append(copy.deepcopy(particles))
+            fieldAppend = {}
+            for resType in resTypes:
+                fieldAppend[resType] = fields[resType]
+            Fields.append(copy.deepcopy(fieldAppend))
```

Comment:
since this is basically 3 times the same code, maybe make a small internal extra function that handles this?
`Fields, Particles = extractOutput(particles, fields,resTypes`
(or somethin like this)

Reply:
done